src/main/java/teammates/ui/output/SessionResultsData.java

```diff
@@ -85,20 +85,31 @@ public static SessionResultsData initForStudent(SessionResultsBundle bundle, Stu
             Map<String, List<ResponseOutput>> otherResponsesMap = new HashMap<>();
 
             if (questionDetails.isIndividualResponsesShownToStudents()) {
+                boolean canStudentSeeResponsesToSelf = question.isResponseVisibleTo(FeedbackParticipantType.RECEIVER)
+                        || question.isResponseVisibleTo(FeedbackParticipantType.STUDENTS)
+                        || question.isResponseVisibleTo(FeedbackParticipantType.OWN_TEAM_MEMBERS)
+                        || question.isResponseVisibleTo(FeedbackParticipantType.RECEIVER_TEAM_MEMBERS);
```

Comment:
I would like to ask if this is ok or is there is a better way to do this. The intention here is to check if self responses should be added to responsesToSelf array. I used this to check if the responses are supposed to be visible to any student or participant (they would have been filtered out beforehand if they are not supposed to be visible to this user).

Reply:
I think you can remove this boolean check and leave the other logic be. I will explain my rationale to you in abit. 

Let me first clarify something with prof @damithc on the issue page: https://github.com/TEAMMATES/teammates/issues/11146

Reply:
@samuelfangjw please leave out the boolean check and leave the logic as is. 

e.g. do this:
```
if (isUserRecipient) {
    qnOutput.responsesToSelf.add(responseOutput);
}

if (isUserGiver) {
    qnOutput.responsesFromSelf.add(responseOutput);
}
```

The reason is that we already preprocess the visibility settings for each question when we obtain the bundle. Look at `getViewableFeedbackResponsesForStudentForQuestion` in `FeedbackResponsesLogic.java`. We don't need to do any more checks from that point onwards.

In the frontend, use the visibility of the question to decide out on the logic of displaying on the UI:

```
"feedbackQuestion":{
            "feedbackQuestionId":"ahBzfnRlYW1tYXRlcy1qb2huchYLEhBGZWVkYmFja1F1ZXN0aW9uGCIM",
            "questionNumber":1,
            "questionBrief":"ewed",
            "questionDescription":"",
            "questionDetails":{
               "minScale":1,
               "maxScale":5,
               "step":0.5,
               "questionType":"NUMSCALE",
               "questionText":"ewed"
            },
            "questionType":"NUMSCALE",
            "giverType":"STUDENTS",
            "recipientType":"OWN_TEAM_MEMBERS_INCLUDING_SELF",
            "numberOfEntitiesToGiveFeedbackToSetting":"UNLIMITED",
            "showResponsesTo":[
               "INSTRUCTORS"
            ],
            "showGiverNameTo":[
               "INSTRUCTORS"
            ],
            "showRecipientNameTo":[
               "INSTRUCTORS"
            ]
         }
```

Look at `showResponsesTo` above. This is a more robust way of doing this. Relying on an empty array is prone to bugs down the road. Let me know if this makes sense or if you have an alternate proposal.

Reply:
@rrtheonlyone Thanks for the review, I'll make the changes soon!